uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Document.php

```diff
@@ -78,7 +79,7 @@ protected static function cloneApprovedDocumentsWithExpireDate($expireDate) {
       . "LEFT JOIN civicrm_option_group og ON og.name = 'activity_type' "
       . "LEFT JOIN civicrm_option_value ov ON ov.option_group_id = og.id AND ov.component_id = c.id "
       . "LEFT JOIN civicrm_activity a ON a.id = ac.entity_id "
-      . "WHERE DATE(ac.expire_date) = %1 AND a.is_deleted = 0 AND a.status_id = 3 AND ov.component_id = c.id AND a.activity_type_id = ov.value";
+      . "WHERE DATE(ac.expire_date) >= CURDATE() AND DATE(ac.expire_date) <= %1 AND ac.clone_date IS NULL AND a.is_deleted = 0 AND a.status_id = 3 AND ov.component_id = c.id AND a.activity_type_id = ov.value";
```

Comment:
Can you please add some tests to add coverage to this change? I can think of a few scenarios that should be tested:
- Test that a document isn't cloned more than once after consecutive calls to the scheduled job
- Test that the clone_date is set properly (You don't really need this if the previous test is working, but it's ok to add it if you feel like doing it)
- Test that the scheduled job will clone anything <= than the current date + 'days_to_create_a_document'